cypress-tests/cypress.config.js

```diff
@@ -28,7 +29,18 @@ export default defineConfig({
           return null;
         },
       });
-
+      on("after:spec", (spec, results) => {
+        if (results && results.video) {
+          // Do we have failures for any retry attempts?
+          const failures = results.tests.some((test) =>
+            test.attempts.some((attempt) => attempt.state === "failed")
+          );
+          if (!failures) {
+            // delete the video if the spec passed and no tests retried
+            fs.unlinkSync(results.video);
+          }
+        }
+      });
```

Comment:
we've enabled video recording in `L63`. Since we do not want to keep the whole recording but only for the tests that fail, this section is necessary to keep the report clean